plugins/Live/API.php

```diff
@@ -207,12 +207,10 @@ public function getVisitorProfile($idSite, $visitorId = false, $segment = false,
             $visitorId = $this->getMostRecentVisitorId($idSite, $segment);
         }
 
-        $newSegment = ($segment === false ? '' : $segment . ';') . 'visitorId==' . $visitorId;
-
         $limit = Config::getInstance()->General['live_visitor_profile_max_visits_to_aggregate'];
 
-        $visits = $this->loadLastVisitorDetailsFromDatabase($idSite, $period = false, $date = false, $newSegment,
-            $offset = 0, $limit);
+        $visits = $this->loadLastVisitorDetailsFromDatabase($idSite, $period = false, $date = false, $segment,
+            $offset = 0, $limit, false, false, $visitorId);
```

Comment:
Not sure why we added the visitorid to the segment instead of setting the parameter, as the used method directly supports it. The resulting query for fetching the visits changes only slightly:

with segment:
```sql
SELECT sub.*
FROM (
SELECT log_visit.*
FROM piwik_log_visit AS log_visit
WHERE (log_visit.idsite IN (?)) AND (log_visit.idvisitor = ?)
ORDER BY idsite DESC, visit_last_action_time DESC
LIMIT 0, 1000) AS sub
GROUP BY sub.idvisit
ORDER BY sub.visit_last_action_time DESC
LIMIT 100;
```

with param:
```sql
SELECT sub.*
FROM (
SELECT log_visit.*
FROM piwik_log_visit AS log_visit
WHERE log_visit.idsite IN (?) AND log_visit.idvisitor = ?
ORDER BY idsite DESC, visit_last_action_time DESC
LIMIT 0, 100) AS sub
GROUP BY sub.idvisit
ORDER BY sub.visit_last_action_time DESC
LIMIT 100;
```

Not sure if that has any impact on performance...

Reply:
There should be no difference in performance as MySQL will parse this the same way IMO 👍

Reply:
There is a slightly difference as the inner query now only has a limit of `100` instead of `1000`, so maybe it might be even faster, as it has lower amount of rows to sort in outer query ;-)